database/src/test/scala/za/co/absa/atum/database/runs/GetPartitioningCheckpointsTest.scala

```diff
@@ -0,0 +1,352 @@
+        .add("fk_checkpoint", checkpointId)
+        .add("measurement_value", measurementAvg)
+    )
+
+    table("runs.measurements").insert(
+      add("fk_measure_definition", measureDefinitionOtherId)
+        .add("fk_checkpoint", checkpointOtherId)
+        .add("measurement_value", measurementSum)
+    )
+
+    val actualMeasures: Seq[MeasuredDetails] = function(fncGetPartitioningCheckpoints)
+      .setParam("i_partitioning", partitioning)
+      .setParam("i_checkpoint_name", "CheckpointNameCntAndAvg")
+      .execute { queryResult =>
+        assert(queryResult.hasNext)
+        val row1 = queryResult.next()
+        assert(row1.getInt("status").contains(11))
+        assert(row1.getString("status_text").contains("Ok"))
+        assert(row1.getUUID("id_checkpoint").contains(checkpointId))
+        assert(row1.getString("checkpoint_name").contains("CheckpointNameCntAndAvg"))
+        assert(row1.getOffsetDateTime("checkpoint_start_time").contains(startTime))
+        assert(row1.getOffsetDateTime("checkpoint_end_time").contains(endTime))
+
+        val measure1 = MeasuredDetails(
+          row1.getString("measure_name").get,
+          row1.getArray[String]("measure_columns").map(_.toList).get,
+          row1.getJsonB("measurement_value").get
+        )
+
+        val row2 = queryResult.next()
+        assert(row2.getInt("status").contains(11))
+        assert(row2.getString("status_text").contains("Ok"))
+        assert(row2.getUUID("id_checkpoint").contains(checkpointId))
+        assert(row2.getString("checkpoint_name").contains("CheckpointNameCntAndAvg"))
+        assert(row2.getOffsetDateTime("checkpoint_start_time").contains(startTime))
+        assert(row2.getOffsetDateTime("checkpoint_end_time").contains(endTime))
+
+        val measure2 = MeasuredDetails(
+          row2.getString("measure_name").get,
+          row2.getArray[String]("measure_columns").map(_.toList).get,
+          row2.getJsonB("measurement_value").get
+        )
+
+        assert(!queryResult.hasNext)
+        Seq(measure1, measure2)
+      }
+
+    assert(actualMeasures.map(_.measureName).toSet == Set("avg", "cnt"))
+    assert(actualMeasures.map(_.measureColumns).toSet == Set(Seq("col1"), Seq("a", "b")))
+    actualMeasures.foreach { currVal =>
+      val currValStr = currVal.measurementValue.value
+      assert(currValStr.contains(""""value": "2.71"""") || currValStr.contains(""""value": "3""""))
+    }
+  }
+
+  test("Get partitioning checkpoints returns no checkpoints for partitioning without checkpoints") {
+
+    table("runs.partitionings").insert(
+      add("partitioning", partitioning2)
+        .add("created_by", "Daniel")
+    )
+
+    function(fncGetPartitioningCheckpoints)
+      .setParam("i_partitioning", partitioning2)
+      .setParam("i_limit", i_limit)
+      .setParam("i_checkpoint_name", i_checkpoint_name)
+      .execute { queryResult =>
+        val results = queryResult.next()
+        assert(results.getInt("status").contains(16))
```

Comment:
hmm in your function you no longer have status 16 right? So this test should fail...?

Reply:
Yeah, I actually missed that. Thank you.